.cbmc-batch/source/utils.c

```diff
@@ -16,15 +16,15 @@
 #include <proof_helpers/utils.h>
 
 void assert_bytes_match(const uint8_t *a, const uint8_t *b, size_t len) {
-    if (len > 0) {
+    if (len > 0 && a != NULL && b != NULL) {
```

Comment:
Do we want to assert that both a and b are null, or both are non-null?

Reply:
If a or b are null inside this block, we can't `assert(a[i] == b[i]);`

Reply:
My point was that in your implementation, if a is NULL but b is not, this function will return success.